src/scale.jl

```diff
@@ -251,45 +253,51 @@ function apply_scale_typed!(ds, field, scale::ContinuousScale)
 end
 
 # Reorder the levels of a pooled data array
-function reorder_levels(da::PooledDataArray, order::AbstractVector)
-    level_values = levels(da)
+function reorder_levels(da::IndirectArray, order::AbstractVector)
+    level_values = da.values
     length(order) != length(level_values) &&
             error("Discrete scale order is not of the same length as the data's levels.")
     permute!(level_values, order)
-    return PooledDataArray(da, level_values)
+    return IndirectArray(da, level_values)
 end
 
-function discretize_make_pda(values::Vector, levels=nothing)
-    if levels == nothing
-        return PooledDataArray(values)
-    else
-        return PooledDataArray(convert(Vector{eltype(levels)}, values), levels)
-    end
-end
+discretize_make_pda(values::Vector)         = IndirectArray(values)
+discretize_make_pda(values::Vector, ::Void) = IndirectArray(values)
+discretize_make_pda(values::Vector, levels) =
+    IndirectArray(map(t -> findfirst(levels, t), values), levels)
 
-function discretize_make_pda(values::DataArray, levels=nothing)
-    if levels == nothing
-        return PooledDataArray(values)
-    else
-        return PooledDataArray(convert(DataArray{eltype(levels)}, values), levels)
-    end
-end
+discretize_make_pda(values::DataArray)         = IndirectArray(values)
```

Comment:
I don't think this specialization for `DataArray` is needed, is it? Even when missing values need special treatment, `AbstractArray{>:Missing}` is usually enough.

Reply:
I'd prefer not to change this now. We can do another round of cleanup in a new PR.

Reply:
It might be possible to define these for more abstract types but I'm not sure it will always do the right thing so, for now, I prefer strict signatures.